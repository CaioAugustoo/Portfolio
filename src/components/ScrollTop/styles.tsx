import styled, { css } from "styled-components";

export const ScrollTop = styled.button`
  ${({ theme }) => css`
    background: ${theme.colors.primary};

    position: fixed;
    right: 15px;
    bottom: 30px;

    padding: 8px 9px;

    border: none;
    outline: none;
    border-radius: 100px;

    cursor: pointer;
    opacity: 0;
    transition: all 0.4s ease;

    &:hover {
      box-shadow: 0 20px 25px -5px rgba(0, 0, 0, 0.1),
        0 10px 10px -5px rgba(0, 0, 0, 0.1);
      transform: translate3d(0px, -3px, 0px);
    }
  `}
`;
