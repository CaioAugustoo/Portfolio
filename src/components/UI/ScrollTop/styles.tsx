import styled, { css } from "styled-components";

export const ScrollTop = styled.button`
  ${({ theme }) => css`

  `}
`;
