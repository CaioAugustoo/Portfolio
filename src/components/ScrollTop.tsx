import { useEffect, useState } from "react";
import * as S from "./styles";
import { FC } from "react";
import { Link } from "react-scroll";

const ScrollTop: FC = () => {
  const [scrolled, setScrolled] = useState(false);

  useEffect(() => {
    const verifyUserScroll = () => {
      window.pageYOffset > 600 ? setScrolled(true) : setScrolled(false);
    };

    window.addEventListener("scroll", verifyUserScroll);
    return () => {
      window.removeEventListener("scroll", verifyUserScroll);
    };
  });

  return (
    <S.ScrollTop style={{ opacity: scrolled ? 1 : 0 }}>
      <Link to="home" spy={true} smooth="easeInOutQuart" duration={1000}>
        <img
          src="/img/back_top.svg"
          alt="Icone para voltar ao topo da imagem"
        />
      </Link>
    </S.ScrollTop>
  );
};

export default ScrollTop;
