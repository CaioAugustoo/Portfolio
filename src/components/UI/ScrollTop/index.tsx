import { useEffect, useState } from "react";
import { Link } from "react-scroll";

import * as S from "./styles";

const ScrollTop = () => {
  const [scrolled, setScrolled] = useState(false);

  useEffect(() => {
    const handler = () => {
      window.pageYOffset > 600 ? setScrolled(true) : setScrolled(false);
    };

    window.addEventListener("scroll", handler);
    return () => window.removeEventListener("scroll", handler);
  }, []);

  return (
    <S.ScrollTop scrolled={scrolled}>
      <Link to="home" spy={true} smooth="easeInOutQuart" duration={1000}>
        <img
          src="/img/back_top.svg"
          alt="Icone para voltar ao topo da imagem"
        />
      </Link>
    </S.ScrollTop>
  );
};

export default ScrollTop;
