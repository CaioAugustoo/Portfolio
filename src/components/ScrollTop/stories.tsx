import ScrollTop from ".";
import { Story, Meta } from "@storybook/react/types-6-0";

export default {
  title: "ScrollTop",
  component: ScrollTop,
} as Meta;

export const Basic: Story = (args) => <ScrollTop {...args} />;
