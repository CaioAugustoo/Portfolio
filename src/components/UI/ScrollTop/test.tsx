import { screen } from "@testing-library/react";
import { renderWithTheme } from "utils/tests/helpers";

import ScrollTop from ".";

describe("<ScrollTop />", () => {
  it("should render correctly", () => {
    const { container } = renderWithTheme(<ScrollTop />);

    expect(
      screen.getByAltText(/Icone para voltar ao topo da imagem/i)
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
