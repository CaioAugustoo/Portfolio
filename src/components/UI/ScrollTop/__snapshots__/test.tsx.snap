// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollTop /> should render correctly 1`] = `
.c0 {
  background: #7962ff;
  position: fixed;
  right: 30px;
  bottom: 30px;
  border-radius: 100px;
  padding: 0.9rem;
  border: none;
  cursor: pointer;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  opacity: 0;
  pointer-events: none;
}

.c0:hover {
  background: #4C30F5;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

<button
  class="c0"
>
  <a>
    <img
      alt="Icone para voltar ao topo da imagem"
      src="/img/back_top.svg"
    />
  </a>
</button>
`;
